### ISSUE TITLE
ci: rbd encryption tests are failing due to vault 

### ISSUE THREAD

**Issue by Madhu-1** [ceph/ceph-csi#2657]
Looks like the e2e tests are broken due to vault issue 

```
StorageClass "csi-rbd-sc": rpc error: code = InvalidArgument desc = invalid encryption kms configuration: failed connecting to Vault: failed to get the authentication token: Put "http://vault.cephcsi-e2e-ab3babb4a6ad.svc.cluster.local:8200/v1/auth/kubernetes/login": dial tcp: lookup vault.cephcsi-e2e-ab3babb4a6ad.svc.cluster.local: no such host
Nov 19 03:13:19.008: INFO: POD  NODE  PHASE  GRACE  CONDITIONS
```

```
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:02:50 +0000 GMT - event for vault-init-job-4tgrs: {default-scheduler } Scheduled: Successfully assigned cephcsi-e2e-ab3babb4a6ad/vault-init-job-4tgrs to minikube
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:02:51 +0000 GMT - event for vault-init-job-4tgrs: {kubelet minikube} Pulled: Container image "docker.io/library/vault:latest" already present on machine
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:02:51 +0000 GMT - event for vault-init-job-4tgrs: {kubelet minikube} Started: Started container vault-init-job
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:02:51 +0000 GMT - event for vault-init-job-4tgrs: {kubelet minikube} Created: Created container vault-init-job
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:08:12 +0000 GMT - event for vault-init-job: {job-controller } SuccessfulCreate: Created pod: vault-init-job-rgm6r
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:08:12 +0000 GMT - event for vault-init-job-rgm6r: {default-scheduler } Scheduled: Successfully assigned cephcsi-e2e-ab3babb4a6ad/vault-init-job-rgm6r to minikube
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:08:13 +0000 GMT - event for vault-init-job-rgm6r: {kubelet minikube} Created: Created container vault-init-job
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:08:13 +0000 GMT - event for vault-init-job-rgm6r: {kubelet minikube} Pulled: Container image "docker.io/library/vault:latest" already present on machine
Nov 19 03:13:19.308: INFO: At 2021-11-19 03:08:13 +0000 GMT - event for vault-init-job-rgm6r: {kubelet minikube} Started: Started container vault-init-job
Nov 19 03:13:19.314: INFO: POD                                            NODE      PHASE    GRACE  CONDITIONS
Nov 19 03:13:19.314: INFO: csi-cephfsplugin-provisioner-7f7bcf4d54-c722t  minikube  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:40 +0000 GMT  } {Ready True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:51 +0000 GMT  } {ContainersReady True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:51 +0000 GMT  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:40 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: csi-cephfsplugin-wvjvq                         minikube  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:40 +0000 GMT  } {Ready True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:50 +0000 GMT  } {ContainersReady True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:50 +0000 GMT  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:51:40 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: csi-rbdplugin-provisioner-6b7864574f-8t5ld     minikube  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:53:56 +0000 GMT  } {Ready True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:54:00 +0000 GMT  } {ContainersReady True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:54:00 +0000 GMT  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:53:56 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: csi-rbdplugin-vbbns                            minikube  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:58:30 +0000 GMT  } {Ready True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:58:32 +0000 GMT  } {ContainersReady True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:58:32 +0000 GMT  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:58:30 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: vault-5bcdf8c6f7-dxfdn                         minikube  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:52:25 +0000 GMT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:55:13 +0000 GMT ContainersNotReady containers with unready status: [monitor]} {ContainersReady False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:55:13 +0000 GMT ContainersNotReady containers with unready status: [monitor]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:52:25 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: vault-init-job-4tgrs                           minikube  Failed          [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:02:50 +0000 GMT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:07:52 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {ContainersReady False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:07:52 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:02:50 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: vault-init-job-5htdl                           minikube  Failed          [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:57:38 +0000 GMT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:02:40 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {ContainersReady False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:02:40 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:57:38 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: vault-init-job-j5rhv                           minikube  Failed          [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:52:35 +0000 GMT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:57:38 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {ContainersReady False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:57:38 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 02:52:35 +0000 GMT  }]
Nov 19 03:13:19.314: INFO: vault-init-job-rgm6r                           minikube  Failed          [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:08:12 +0000 GMT  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:13:14 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {ContainersReady False 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:13:14 +0000 GMT ContainersNotReady containers with unready status: [vault-init-job]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2021-11-19 03:08:12 +0000 GMT  }]
```

vault job is not in ready state.

**Comment by Madhu-1** [comment]
@ceph/ceph-csi-contributors until this is fixed CI is broken for now. retest of any PR's won't help. please avoid retesting the  PR's

**Comment by Madhu-1** [comment]
fixed in #2658 closing
